File: P1S2/VacunacionFactory/CiudadanoModerna.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package VacunacionFactory;

import java.util.Random;

/**
 *
 * @author ines
 */
public class CiudadanoModerna extends Ciudadano {
    
    public CiudadanoModerna(){
        Random random  = new Random();
        this.dni = random.nextInt(99999999+1);
        //Ciudadano con un rango de edad entre 50 y 110
        this.edad = random.nextInt(110+1-50)+50;
        
         System.out.println("Mi DNI es "+ dni+ ", tengo " + edad + 
         " años me voy a vacunar de Moderna :D \n");
    }
    
    public void getTipoCiudadano(){
        System.out.println("Mi DNI es " + dni +", tengo " + edad + 
        " años y me he vacunado de Moderna ;)\n");
    }
    
}

File: P1/P1S2/Ruby/VacunacionFactoriaPrototipo/ciudadano_moderna.rb
#encoding: UTF-8

require_relative 'ciudadano'

module Vacunacion
  class CiudadanoModerna < Ciudadano
    def initialize
      super
      @dni = rand(100000000..999999999)
      @edad = rand(50..111)
    end

    def toString
      puts("Mi DNI es #{@dni}, tengo #{@edad} y me he vacunado de Moderna :D")
    end
  end
end